build.rs

```diff
@@ -43,4 +43,9 @@ fn main() {
             "The `jtag_serial` feature is only supported by the ESP32-C3, ESP32-C6, ESP32-H2, and ESP32-S3"
         );
     }
+
+    // Ensure that, if the `colors` is used with `log`.`
+    if cfg!(feature = "colors") && !cfg!(feature = "log") {
+        panic!("The `colors` feature is only effective when using the `log` feature");
```

Comment:
```suggestion
        println!("cargo:warning=The `colors` feature is only effective when using the `log` feature");
```

We should change this to print a warning rather than panicking.